pkg/types/chain_reader.go

```diff
@@ -46,6 +46,12 @@ type ChainReader interface {
 	// Similarly, when using a struct for returnVal, fields in the return value that are not on-chain will not be set.
 	GetLatestValue(ctx context.Context, contractName, method string, params, returnVal any) error
 
+	// BatchGetLatestValue batches get latest value calls based on request, which is grouped by contract names that each have a slice of BatchRead.
+	// BatchRead params and returnVal follow same rules as GetLatestValue params and returnVal arguments, with difference in how response is returned.
+	// BatchGetLatestValueResult response is grouped by contract names, which contain read results that maintain the order from the request.
+	// Contract call errors are returned in the Err field of BatchReadResult.
+	BatchGetLatestValue(ctx context.Context, request BatchGetLatestValueRequest) (BatchGetLatestValueResult, error)
```

Comment:
The name feels a little clunky to me. Does any of these read more intuitively?
```suggestion
	GetBatchLatestValue(ctx context.Context, request GetBatchLatestValueRequest) (GetBatchLatestValueResult, error)
```
```suggestion
	GetLatestValueBatch(ctx context.Context, request GetLatestValueBatchRequest) (GetLatestValueBatchResult, error)
```
```suggestion
	GetLatestValues(ctx context.Context, request GetLatestValuesRequest) (GetLatestValuesResult, error)
```

Reply:
`BatchGetLatestValue` and `GetLatestValue` mimic `BatchCall` and `Call`, so the current naming makes sense to me because of that. If you have a strong opinion I would at least keep Batch in the name, `GetLatestValueBatch` could work

Reply:
Are those the eth client names? We need to be careful not to let that influence the agnostic API too much.

@krehermann where is that google naming guide? Does it cover list kind of multivalue request? because 'List' doesn't really fit.

Reply:
I tend to like the plural version and it works well with code completion: `GetLatestValue_`.

Reply:
Im okay with `GetLatestValues`

Reply:
IMHO `Values` is the most ambiguous option, and something we might want to use later for a different purpose.

Reply:
tbh `GetLatestValueBatch` and `GetBatchLatestValue` don't feel less clunky than `BatchGetLatestValue`, but I am not too defensive

Reply:
I'm worried about presuming we will only need one plural form, but maybe that isn't a realistic concern :shrug:

Reply:
Changed to `BatchGetLatestValues`